protocol/storage/client.go

```diff
@@ -179,8 +176,8 @@ func (smc *Client) ProposeDeal(ctx context.Context, miner address.Address, data
 		if err != nil {
 			return nil, err
 		}
-	case <-ctxSetup.Done():
-		return nil, ctxSetup.Err()
+	case <-time.After(30 * time.Second):
```

Comment:
There is already a timeout at line 104, the channel will hold an error if it expires. This additional timeout is confusing and I don't think necessary. I think we only want a simple read from the channel.

Reply:
ah interesting, makes this even simpler.